R/clusterSystems.R

```diff
@@ -0,0 +1,146 @@
+#'
+#' @section Custom distance metrics: In addition to several distance metrics which are built in to the function, the user has the option of defining a new distance metric to measure pairwise similarity between genes. If this is the case, an appropriate source of data (mapping from HGNC symbols to data of interest) that can be used as in input for each custom distance function.
+#'
+#' @param systems The input systems (lists of HGNC symbols) that should be re-clustered. Each element is formatted according to the output of \code{SyDBgetSysSymbols}.
+#' @param distances Character vector indicating choice of data to be used to measure distance between input genes. Vector containing one of more of \code{c("expression_profile", "transcription_factor", "network_jaccard", "network_distance")} or NULL if a custom distance metric is to be used instead.
+#' @param customDistanceFn Optional list of functions. Any custom functions that are to be used to calculate pairwise distances between genes. Default is NULL. If this parameter is not NULL, then \code{dataSources} parameter parameter must also be provided and be a list of the same length.
+#' @param dataSources Optional list of input data appropriate to be used in conjunction with \code{customDistanceFn}. The entries of this list should be in the same order as \code{customDistanceFn} such that the ith element of \code{customDistanceFn} uses the data provided in the ith element of \code{dataSources} as input. Default is NULL.
+#' @param combineMatrices String indicating the method by which to combine distance matrices if more than one choice of distance data is provided. If only one type of distance data is provided, then this parameter is ignored. One of \code{c('sum', 'product', 'minimum', 'maximum')}.
+#' @param printVennDiagrams Logical flag; indicated whether or not Venn diagrams representing the set overlap of the output should be printed when the function is called. Default is TRUE.
+#'
+#' @return A named list of length 4. Elements of the list include:
+#'
+#' \code{$Clusters} A named vector of integers from 1 to k where k is the number of clusters. Names of the elements represent the genes belonging to that cluster.
+#'
+#' \code{$Best_matches} A named vector of length k representing the cluster with the best set overlap for each input system.
+#'
+#' \code{$Jaccard_indexes} The Jaccard indexes measuring the similarity between each system and the cluster which is its best match.
+#'
+#' \code{$P_values} P values representing the probability of having a Jaccard index greater than those observed by choosing a random cluster of the same size from the set of input genes.
+#'
+#' @author \href{https://orcid.org/0000-0001-5724-2252}{Rachel Silverstein} (aut)
+#'
+#' @seealso For examples of distance functions that can be used to create custom distance matrices, see \code{\link{expr_dist}} \code{\link{jaccard_dist}} \code{\link{tf_dist}}.
+#'
+#' @examples
+#' systemDB_name <- "SysDB"
```

Comment:
You can use `"SysDB"` directly, there's nothing gained here by intermediate assignment to `systemDB_name`.